monitorcontrol/vcp/vcp_linux.py

```diff
@@ -390,12 +390,20 @@ def get_vcps() -> List[LinuxVCP]:
 
     # iterate I2C devices
     for device in pyudev.Context().list_devices(subsystem="i2c"):
-        vcp = LinuxVCP(device.sys_number)
         try:
+            if sys_no := device.sys_number:
+                vcp = LinuxVCP(int(sys_no))
+            else:
+                logging.error(
+                    "Unable to check i2c device %s: no device number found", device
+                )
+                continue
```

Comment:
I am curious, have you seen this occur?  `pyudev` should only return `None` if the device number is not a unicode string: https://github.com/pyudev/pyudev/blob/7588bc8c84bf1e994b31020452124bc6618625ce/src/pyudev/device/_device.py#L626-L651